pkg/tests/integration/test_salt_user.py

```diff
@@ -170,3 +171,191 @@ def test_pkg_paths(
+                                f"{install_salt.conf_dir}/master",
+                                "user: salt",
+                                f"user: {test_account.username}",
+                                "flags=['IGNORECASE']",
+                                "append_if_not_found=True",
+                            )
+                            assert ret.returncode == 0
+
+                            # change ownership of appropriate paths to user
+                            for _path in log_pkg_paths:
+                                chg_ownership_cmd = (
+                                    f"chown -R {test_account.username} {_path}"
+                                )
+                                ret = salt_call_cli.run(
+                                    "--local", "cmd.run", chg_ownership_cmd
+                                )
+                                assert ret.returncode == 0
+
+                            # restart the salt_master
+                            with salt_master.started():
+                                assert salt_master.is_running() is True
+
+                                # ensure some data in files
+                                log_files_list = [
+                                    "/var/log/salt/api",
+                                    "/var/log/salt/key",
+                                    "/var/log/salt/master",
+                                ]
+                                for _path in log_files_list:
+                                    log_path = pathlib.Path(_path)
+                                    assert log_path.exists()
+                                    with log_path.open("a") as f:
+                                        f.write("This is a log rotation test\n")
+
+                                # force log rotation
+                                logr_conf_file = "/etc/logrotate.d/salt"
+                                logr_conf_path = pathlib.Path(logr_conf_file)
+                                if not logr_conf_path.exists():
+                                    logr_conf_file = "/etc/logrotate.conf"
+                                    logr_conf_path = pathlib.Path(logr_conf_file)
+                                    assert logr_conf_path.exists()
+
+                                # force log rotation
+                                log_rotate_cmd = f"logrotate -f  {str(logr_conf_file)}"
+                                ret = salt_call_cli.run(
+                                    "--local", "cmd.run", log_rotate_cmd
+                                )
+                                assert ret.returncode == 0
+
+                                for _path in log_files_list:
+                                    log_path = pathlib.Path(_path)
+                                    assert log_path.exists()
+                                    assert (
+                                        log_path.owner() == f"{test_account.username}"
```

Comment:
```suggestion
                                        log_path.owner() == test_account.username
```
Why do you need this cast to a string? `test_account.username` is already a string.

Reply:
bit of left over debug I missed after I deleted the debug logging etc during development, got rid of all the debug lines etc. but missed that f-string conversion :facepalm: , it was getting printed out in a log message, and just had a couple of f-string conversions, and missed that last one when cleaning up

Reply:
updated